protocols/v2/channels-sv2/src/server/extended.rs

```diff
@@ -1,4 +1,47 @@
-//! Mining Server abstraction over the state of a Sv2 Extended Channel
+//! # SV2 Extended Channel - Mining Server Abstraction.
+//!
+//! This module defines the [`ExtendedChannel`] struct, which provides an abstraction of a Stratum
+//! V2 (SV2) extended channel as maintained by a mining pool server.
+//!
+//! An extended channel represents a logical mining connection between a pool and a downstream
+//! mining entity, such as a mining device or proxy, that supports version rolling, extranonce
+//! management, and share validation over multiple jobs.
+//!
+//! ## Responsibilities
+//!
+//! `ExtendedChannel` is responsible for managing all the state associated with an SV2 extended
+//! channel, including:
+//!
+//! - **Channel Parameters**: Holds the unique `channel_id`, `user_identity`, `extranonce_prefix`,
+//!   and other parameters negotiated during channel opening.
+//! - **Target Difficulty**: Manages the target difficulty (`target`) and maximum allowed target
+//!   (`requested_max_target`), based on client requests and nominal hashrate.
+//! - **Job Lifecycle Management**: Stores jobs received from new templates or custom job messages,
+//!   including:
+//!   - Future jobs (indexed by `template_id`)
+//!   - Active job (currently being mined)
+//!   - Past and stale jobs (for share validation over time)
+//! - **Share Validation and Accounting**: Validates shares submitted by the miner, updating
+//!   internal accounting and detecting duplicates or stale submissions. Determines if a share meets
+//!   the channel or network target and responds accordingly.
+//! - **Chain Tip Management**: Tracks the latest known chain tip (block height, previous hash,
+//!   timestamp, and target) for constructing headers and validating shares.
+//! - **Version Rolling**: Honors client configuration on whether version rolling is permitted,
+//!   validating submitted block versions accordingly.
```

Comment:
```suggestion
//! - **Version Rolling**: Honors server configuration on whether version rolling is permitted,
//!   validating submitted BIP320 header versions accordingly.
```

this is server configuration, not client

yes, client can set `SetupConnection.REQUIRES_VERSION_ROLLING` but it's up to the server to allow that or not